src/engine.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <string>
+#include <functional>
+
+#include "status.h"
+
+namespace pmem {
+namespace kv {
+
+using status = pmemkv_status;
+
+typedef void(AllCallback)(void* context, int keybytes, const char* key);
```

Comment:
"void *context" and "const char *key"

`*` belongs to a variable - if it would belong to the type this:
```
char* ptr1, ptr2;
```
would declare two pointers.